plugins/pulp_ostree/plugins/lib.py

```diff
@@ -360,9 +394,15 @@ def list_refs(self):
         _list = []
         lib = Lib()
         self.open()
-        flags = lib.OSTree.RepoPullFlags.COMMIT_ONLY
-        _, summary = self.impl.remote_list_refs(self.id, None)
+        try:
+            _, summary = self.impl.remote_list_refs(self.id, None)
+        except lib.GLib.GError:
+            if not required:
+                summary = {}
+            else:
+                raise
```

Comment:
I feel a bit silly now because I got to the test section, saw there was an assertion about the exception type, and when I ran the tests they all passed. Then I remembered the decorator.